src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -200,11 +200,11 @@ private async Task ProcessQueueAsync()
                     else
                     {
                         // Non mutating are fire-and-forget because they are by definition readonly. Any errors
-                        // will be sent back to the client but we can still capture errors in queue processing
-                        // via NFW, though these errors don't put us into a bad state as far as the rest of the queue goes.
+                        // will be sent back to the client but they can also be captured via HandleNonMutatingRequestError,
+                        // though these errors don't put us into a bad state as far as the rest of the queue goes.
                         // Furthermore we use Task.Run here to protect ourselves against synchronous execution of work
                         // blocking the request queue for longer periods of time (it enforces parallelizabilty).
-                        _ = Task.Run(() => work.StartRequestAsync(context, cancellationToken), cancellationToken);
+                        _ = Task.Run(() => HandleNonMutatingRequestErrorAsync(work.StartRequestAsync(context, cancellationToken)), cancellationToken);
```

Comment:
In [the previous code](https://github.com/dotnet/roslyn/blob/release/dev17.4/src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.cs#L220) the ReportNonFatal was on the task returned from Task.Run. Not sure if that matters though??

Also.. might be worth checking if there are other spots where we're missing ReportNonFatal, and make this method name a bit more generic. It might be luck that I happened to notice this one, there could be others.

Even if there aren't though, I'm not sure the current name makes sense. Its not handling an error, its just wrapping a task. Perhaps a `WrapStartRequestTask(Task t, bool mutating)` would API would be a little more descriptive?